### PR TITLE
Prefix assets

### PR DESCRIPTION
**Pull request by jpslav**
Optionally prefix assets

### CODE REVIEW

File: lib/openstax/path_prefixer/action_controller_base_extensions.rb [hunk 1]
```diff
@@ -23,4 +24,13 @@ def _compute_redirect_to_location(request, options) #:nodoc:
...
   end
 
+  alias_method :original_url_options, :url_options
+  def url_options
+    # Force the script_name to be in the URL options.  There are some times when
+    # Rails chooses not to, for some inexplicable reasons.  This was added due
+    # to real testing in openstax/accounts; not particularly spec'd here.
+    original_url_options.dup.tap{|options| options[:script_name] = request.script_name}
+  end
+
```
Comment (nathanstitt) [inline lib/openstax/path_prefixer/action_controller_base_extensions.rb:34]: I'm assuming this needs to support Rails 4.x?  if not you should probably use `Module.prepend`.  That's the newfangled way of doing this https://www.justinweiss.com/articles/rails-5-module-number-prepend-and-the-end-of-alias-method-chain/

File: lib/openstax_path_prefixer.rb [hunk 2]
```diff
@@ -14,6 +14,12 @@ def self.configuration
 
     class Configuration
       attr_accessor :prefix
+      attr_accessor :prefix_assets
+
+      def initialize
+        @prefix = ""
+        @prefix_assets = Rails.env.test? ? true : false
+      end
```
Comment (nathanstitt) [inline lib/openstax_path_prefixer.rb:22]: this is enabling it in `development` mode right?  But above in the README, you're suggesting 
`config.prefix_assets = Rails.env.production?` 

Should the two behaviors match?

File: lib/openstax/path_prefixer/action_controller_base_extensions.rb (hunk 1 above)
Comment (jpslav) [reply lib/openstax/path_prefixer/action_controller_base_extensions.rb:34]: Yes, for Accounts.  Good to know about prepend tho.

File: lib/openstax_path_prefixer.rb (hunk 2 above)
Comment (jpslav) [reply lib/openstax_path_prefixer.rb:22]: This is just enabling it in `test` mode so that I can test that it works.

**Pull request merged**